### PR TITLE
Add Axios

### PR DESCRIPTION
**Pull request by jackblanc**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/696386879/pulses/761814654)

We want standardized API access logic on the frontend. 

## This PR

Adds a custom Axios Instance, along with a Token Service for access token management and an API file for API calls. 

Work in progress still, but I think I'd like to merge this and keep working.

### CODE REVIEW

File: src/auth/authAPI.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  SIGNUP = '/api/v1/user/signup/',
+  REFRESH = '/api/v1/user/login/refresh/',
+}
+
+interface LoginRequest {
+  readonly email: string;
+  readonly password: string;
+}
+
+export const login = async (user: LoginRequest) =>
```
Comment (rymaju) [inline src/auth/authAPI.tsx:15]: Huh, did not know you can have multiple exports like this, always though you could only have one.

File: src/auth/axios.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosError, AxiosInstance, AxiosRequestConfig } from 'axios';
+import tokenService from './token';
+import { refresh, logout } from './authAPI';
+
+const Instance: AxiosInstance = axios.create({
+  baseURL: process.env.REACT_APP_API_DOMAIN,
+  timeout: 10000,
```
Comment (rymaju) [inline src/auth/axios.tsx:7]: Should this be a constant or part of some properties file? Just a thought, dont think its a big deal either way.

File: src/auth/axios.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+const requestInterceptor: (
+  request: AxiosRequestConfig,
+) => AxiosRequestConfig = (request: AxiosRequestConfig) => {
+  // eslint-disable-next-line
```
Comment (rymaju) [inline src/auth/axios.tsx:16]: classic

File: src/auth/token.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+enum TOKEN_KEY {
```
Comment (rymaju) [inline src/auth/token.tsx:1]: This all looks really good. 👍

**Review by rymaju — COMMENTED** [review]
Looks like it isnt passing yet, but looks great so far.

**Review by rymaju — APPROVED** [review]

File: src/auth/authAPI.tsx (hunk 1 above)
Comment (jackblanc) [reply src/auth/authAPI.tsx:15]: You can only have one `export default ______`, you can have as many exports as you want though. The difference affects importing: 
`import Thing from 'thing'` will import the default export
`import { login } from './authAPI.tsx` will import this "named export"

File: src/auth/axios.tsx (hunk 2 above)
Comment (jackblanc) [reply src/auth/axios.tsx:7]: Hmmm, yes, going to make it super long on dev to account for debugging active requests.

**Pull request merged**